hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -382,50 +386,56 @@ private void logGroupPasswordInfo() {
-        diagnostics.register(
-                new SystemPropertiesPlugin(loggingService.getLogger(SystemPropertiesPlugin.class)));
-
-        // periodic loggers
-        diagnostics.register(
-                new MetricsPlugin(loggingService.getLogger(MetricsPlugin.class), metricsRegistry, properties));
-        diagnostics.register(
-                new SystemLogPlugin(properties, connectionManager, this, loggingService.getLogger(SystemLogPlugin.class)));
-        diagnostics.register(
-                new NetworkingImbalancePlugin(properties, connectionManager.getNetworking(),
-                        loggingService.getLogger(NetworkingImbalancePlugin.class)));
-        diagnostics.register(
-                new EventQueuePlugin(loggingService.getLogger(EventQueuePlugin.class), listenerService.getEventExecutor(),
-                        properties));
-
-        metricsRegistry.collectMetrics(listenerService);
-
-        proxyManager.init(config, clientContext);
-        listenerService.start();
-        loadBalancer.init(getCluster(), config);
-        partitionService.start();
-        statistics.start();
-        clientExtension.afterStart(this);
-        cpSubsystem.init(clientContext);
+            diagnostics.start();
+
+            // static loggers at beginning of file
+            diagnostics.register(
+                    new BuildInfoPlugin(loggingService.getLogger(BuildInfoPlugin.class)));
+            diagnostics.register(
+                    new ConfigPropertiesPlugin(loggingService.getLogger(ConfigPropertiesPlugin.class), properties));
+            diagnostics.register(
+                    new SystemPropertiesPlugin(loggingService.getLogger(SystemPropertiesPlugin.class)));
+
+            // periodic loggers
+            diagnostics.register(
+                    new MetricsPlugin(loggingService.getLogger(MetricsPlugin.class), metricsRegistry, properties));
+            diagnostics.register(
+                    new SystemLogPlugin(properties, connectionManager, this, loggingService.getLogger(SystemLogPlugin.class)));
+            diagnostics.register(
+                    new NetworkingImbalancePlugin(properties, connectionManager.getNetworking(),
+                            loggingService.getLogger(NetworkingImbalancePlugin.class)));
+            diagnostics.register(
+                    new EventQueuePlugin(loggingService.getLogger(EventQueuePlugin.class), listenerService.getEventExecutor(),
+                            properties));
+
+            metricsRegistry.collectMetrics(listenerService);
+
+            proxyManager.init(config, clientContext);
+            listenerService.start();
+            loadBalancer.init(getCluster(), config);
+            partitionService.start();
+            statistics.start();
+            clientExtension.afterStart(this);
+            cpSubsystem.init(clientContext);
+        } catch (Throwable e) {
+            try {
```

Comment:
Wouldn't it be easier to let the lifecycleService.terminate deal with exception catching and logging. This way the logic remains contained in that class instead of needing to add the complexity here.

Reply:
I actually tried to apply the same as being done for the server side here https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java#L152  Do we really need to log the shutdown failure?

Reply:
Good question. I won't sleep a minute shorter if there would not be a log statement.